packages/peregrine/lib/talons/CartPage/PriceSummary/usePriceSummary.js

```diff
@@ -34,6 +34,7 @@ export const usePriceSummary = props => {
     const isCheckout = !!match;
 
     const { error, loading, data } = useQuery(getPriceSummary, {
+        fetchPolicy: 'cache-and-network',
```

Comment:
Is there a point to doing `cache-and-network` if we never render the cached result? I think this needs some UX treatment (render cache result while we true up with network), or we might as well just do `network-only`. This pattern didn't exist at the time we did `CartPage`, so willing to accept since it does solve the bug.

Reply:
> Is there a point to doing `cache-and-network` if we never render the cached result.

Well, we do render the cached result, and we want to make sure to render changes to it if something else on the page modifies it, especially since we're not using `refetchQueries`.

What results is that the first time we render price summary it will render cache values if it has it and then true-up with the network. If a product is then removed or added the fragment is refetched (that's the pattern we're going with for mutations) updating the cache values which updates the data response here.

> I think this needs some UX treatment (render cache result while we true up with network),

For what it's worth, we already do this, but through a different method - the price summary component takes an `isUpdating` prop and applies a class to the price entries. I don't think we need to do anything.

Reply:
![](https://i.gyazo.com/fa874c3e0e7026b0d14517ebdd9dadd0.gif)

https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/CartPage/PriceSummary/priceSummary.js#L57L59

Reply:
Oh....ok :)

Reply:
I updated it to render the "updating" state during the network fetch rather than `null`.

[![Image from Gyazo](https://i.gyazo.com/92440932926e39d206a2dfbe486aec2e.gif)](https://gyazo.com/92440932926e39d206a2dfbe486aec2e)